6_nodejs/week3/app.js

```diff
@@ -0,0 +1,56 @@
+import knex from "knex";
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "2424",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import express from "express";
+const app = express();
+const port = process.env.PORT || 3002;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knex.select("*").from("contacts");
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort.toString();
+    const validFields = ["first_name", "last_name"]; // Define valid sorting fields
+    const validDirections = ["ASC", "DESC"]; // Define valid sorting directions
```

Comment:
Great job, I see you are solving the SQL inejction via [allowlisting the parameters](https://planetscale.com/blog/how-to-prevent-sql-injection-attacks-in-node-js#allowlisting). That is definitely a great solution, as you can NEVER trust user input. 🧠